Add `ConnectionResetError` as retryable error to `rapids-conda-retry`

This is a mamba error that can be seen here:
https://github.com/rapidsai/cugraph/actions/runs/8175477154/job/22355360420?pr=4215#step:7:66
```
("Connection broken: ConnectionResetError(104, 'Connection reset by peer')", ConnectionResetError(104, 'Connection reset by peer'))
("Connection broken: ConnectionResetError(104, 'Connection reset by peer')", ConnectionResetError(104, 'Connection reset by peer'))


Downloading and Extracting Packages: ...working... done
[rapids-conda-retry] conda returned exit code: 1
[rapids-conda-retry] Exiting, no retryable mamba errors detected: 'ChecksumMismatchError:', 'ChunkedEncodingError:', 'CondaHTTPError:', 'CondaMultiError:', 'ConnectionError:', 'EOFError:', 'JSONDecodeError:', 'Multi-download failed', 'Timeout was reached', segfault exit code 139
[rapids-conda-retry] 
Error: Process completed with exit code 1.
```
I think `ConnectionResetError` should be made retryable here:
https://github.com/rapidsai/gha-tools/blob/main/tools/rapids-conda-retry

CC @rlratzel

--- a/tools/rapids-conda-retry
+++ b/tools/rapids-conda-retry
@@ -90,7 +90,7 @@ function runConda {
             retryingMsg="Retrying after cleaning tarball cache, found 'CondaMultiError:' in output..."
             needToRetry=1
             needToClean=1
-        elif grep -q "Connection broken:": "${outfile}"; then
+        elif grep -q "Connection broken:" "${outfile}"; then
             retryingMsg="Retrying, found 'Connection broken:' in output..."
             needToRetry=1
         elif grep -q ConnectionError: "${outfile}"; then

--- a/tools/rapids-conda-retry
+++ b/tools/rapids-conda-retry
@@ -77,27 +77,30 @@ function runConda {
         # Show exit code
         rapids-echo-stderr "conda returned exit code: ${exitcode}"
 
-        if grep -q CondaHTTPError: "${outfile}"; then
-            retryingMsg="Retrying, found 'CondaHTTPError:' in output..."
-            needToRetry=1
-        elif grep -q ChecksumMismatchError: "${outfile}"; then
+        if grep -q ChecksumMismatchError: "${outfile}"; then
             retryingMsg="Retrying, found 'ChecksumMismatchError:' in output..."
-            needToRetry=1
-        elif grep -q JSONDecodeError: "${outfile}"; then
-            retryingMsg="Retrying, found 'JSONDecodeError:' in output..."
             needToRetry=1
         elif grep -q ChunkedEncodingError: "${outfile}"; then
             retryingMsg="Retrying, found 'ChunkedEncodingError:' in output..."
+            needToRetry=1
+        elif grep -q CondaHTTPError: "${outfile}"; then
+            retryingMsg="Retrying, found 'CondaHTTPError:' in output..."
             needToRetry=1
         elif grep -q CondaMultiError: "${outfile}"; then
             retryingMsg="Retrying after cleaning tarball cache, found 'CondaMultiError:' in output..."
             needToRetry=1
             needToClean=1
-        elif grep -q EOFError: "${outfile}"; then
-            retryingMsg="Retrying, found 'EOFError:' in output..."
+        elif grep -q "Connection broken:": "${outfile}"; then
+            retryingMsg="Retrying, found 'Connection broken:' in output..."
             needToRetry=1
         elif grep -q ConnectionError: "${outfile}"; then
             retryingMsg="Retrying, found 'ConnectionError:' in output..."
+            needToRetry=1
+        elif grep -q EOFError: "${outfile}"; then
+            retryingMsg="Retrying, found 'EOFError:' in output..."
+            needToRetry=1
+        elif grep -q JSONDecodeError: "${outfile}"; then
+            retryingMsg="Retrying, found 'JSONDecodeError:' in output..."
             needToRetry=1
         elif grep -q "Multi-download failed" "${outfile}"; then
             retryingMsg="Retrying, found 'Multi-download failed' in output..."
@@ -115,6 +118,7 @@ function runConda {
 'ChunkedEncodingError:', \
 'CondaHTTPError:', \
 'CondaMultiError:', \
+'Connection broken:', \
 'ConnectionError:', \
 'EOFError:', \
 'JSONDecodeError:', \

